### ISSUE TITLE
final package/dependency structure

### ISSUE THREAD

**Issue by tomwayson** [Esri/cedar#280]
I think we should move most (if not all) cedar-utils code into cedar

**Comment by tomwayson** [comment]
What has complicated this is that we are taking a _potential_ future enhancement (engines) into consideration in our v1 package structure. There's no reason at the current time to have cedar-amcharts in it's own package b/c it's currently the only engine and is required for cedar to work at all. I'm not necesarily saying that we should undo the investment in the monorepo, etc, b/c we _might_ end up having multiple packages. However, I do think that for now we should hide the fact that cedar-amcharts is it's own package from consuming applications.

What I mean is, the current dependency structure (cedar depends on cedar-amcharts) and build output (cedar-amcharts is external and not bundled w/ cedar) will be training users to use a certain pattern (load the engine first from a seperate package). It looks like this when using the UMD build:

```
<!-- in this case we're using the CDNs, but these scripts could point to local files instead -->
<!-- 
  first load the external charting library
  for v1 this will always be amCharts, but in the future
  it would be based on which engine(s) the app will use
--> 
<script src="https://www.amcharts.com/lib/3/amcharts.js"></script>
<script src="https://www.amcharts.com/lib/3/serial.js"></script>
<!-- now load which ever engine you are using, for v1 this is always amCharts -->
<script src="https://unpkg.com/@esri-amcharts/cedar/dist/cedar-amcharts.js"></script>
<!-- now load the cedar base library -->
<script src="https://unpkg.com/@esri/cedar/dist/cedar.js"></script>
```

Or if installing from the registry for use in a custom local build:

```
# NOTE: consuming apps can chose whether or not to treat amCharts as external
npm i amcharts3 @esri/cedar-amcharts @esri/cedar
```

Then you would include these lines in the consuming app:

```
import AmCharts from 'amcharts3'
import Cedar from '@esri/cedar';
// consuming apps don't use any code from cedar-amcharts directly, so they don't import it
```

The problem is we don't know what the actual dependency structure will look like when we introduce a second engine like 'cedar-vega'. I can think of at least 2 viable ways (and several bad ways) that we might arrange dependencies in our multi-engine future, and neither of those two viable ways would use the above loading pattern. For a detailed discussion of what I think those patterns would look like see https://github.com/Esri/cedar/issues/294, but **the TLDR: is, until we figure that out, I think we should try to make it look to consuming apps as though cedar v1 is just one package.** This is consistent w/ the way consumers are used to loading v0, and doesn't start teaching any new loading patterns that we don't know if we'll end up using.

In practice this means leaving the current dependency structure, but bundling cedar-amcharts with cedar, then apps consuming the UMD build would:

```
<!-- in this case we're using the CDNs, but these scripts could point to local files instead -->
<!-- 
  first load the external charting library
  for v1 this will always be amCharts, but in the future
  it would be based on which engine(s) the app will use
--> 
<script src="https://www.amcharts.com/lib/3/amcharts.js"></script>
<script src="https://www.amcharts.com/lib/3/serial.js"></script>
<!-- 
  then just load cedar, which bundles cedar-amcharts at v1
-->
<script src="https://unpkg.com/@esri/cedar/dist/cedar.js"></script>
```

Or if installing from the registry for use in a local build:

```
# NOTE: consuming apps can chose whether or not to treat amCharts as external
# NOTE: @esri/cedar-amcharts will get installed as dep of @esri/cedar
npm i amcharts3 @esri/cedar
```

Then in their code:

```
import AmCharts from 'amcharts3'
import Cedar from '@esri/cedar'
```

**Comment by tomwayson** [comment]
resolved in #299
